[S3] Removing bucket website fails

When running `TestAccS3Bucket_Website_Simple` and other website s3 bucket tests, the following error is shown:
```
    resource_opentelekomcloud_s3_bucket_test.go:189: Step 3/3 error: Error running apply: exit status 1
        
        Error: cannot specify more than one website
        
          with opentelekomcloud_s3_bucket.bucket,
          on terraform_plugin_test.tf line 2, in resource "opentelekomcloud_s3_bucket" "bucket":
           2: resource "opentelekomcloud_s3_bucket" "bucket" {
        
--- FAIL: TestAccS3Bucket_Website_Simple (318.08s)
```

It happens when configuration changes from 
```hcl
resource "opentelekomcloud_s3_bucket" "bucket" {
  bucket = "tf-test-bucket-..."
  acl    = "public-read"

  website {
    index_document = "index.html"
  }
}
```
to 
```hcl
resource "opentelekomcloud_s3_bucket" "bucket" {
  bucket = "tf-test-bucket-%d"
  acl    = "public-read"
}
```

--- a/opentelekomcloud/acceptance/common/common.go
+++ b/opentelekomcloud/acceptance/common/common.go
@@ -13,7 +13,10 @@ import (
 	"github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/common/cfg"
 )
 
-const AlternativeProviderAlias = "opentelekomcloud.alternative"
+const (
+	AlternativeProviderAlias           = "opentelekomcloud.alternative"
+	AlternativeProviderWithRegionAlias = "opentelekomcloud.region"
+)
 
 var (
 	TestAccProviderFactories map[string]func() (*schema.Provider, error)
@@ -31,10 +34,18 @@ provider opentelekomcloud {
   tenant_name = "%s"
 }
 `, altCloud, altProjectID, altProjectName)
+	AlternativeProviderWithRegionConfig string
 )
 
 func init() {
 	TestAccProvider = opentelekomcloud.Provider()
+
+	err := TestAccProvider.Configure(context.Background(), terraform.NewResourceConfigRaw(nil))
+	if err == nil {
+		config := TestAccProvider.Meta().(*cfg.Config)
+		env.OS_REGION_NAME = config.GetRegion(nil)
+	}
+
 	TestAccProviderFactories = map[string]func() (*schema.Provider, error){
 		"opentelekomcloud": func() (*schema.Provider, error) {
 			return TestAccProvider, nil
@@ -50,13 +61,25 @@ func init() {
 			})
 			return provider, nil
 		},
+		"opentelekomcloudregion": func() (*schema.Provider, error) {
+			provider := opentelekomcloud.Provider()
+			provider.Configure(context.Background(), &terraform.ResourceConfig{
+				Config: map[string]interface{}{
+					"cloud":  altCloud,
+					"region": env.OS_REGION_NAME,
+				},
+			})
+			return provider, nil
+		},
 	}
 
-	err := TestAccProvider.Configure(context.Background(), terraform.NewResourceConfigRaw(nil))
-	if err == nil {
-		config := TestAccProvider.Meta().(*cfg.Config)
-		env.OS_REGION_NAME = config.GetRegion(nil)
-	}
+	AlternativeProviderWithRegionConfig = fmt.Sprintf(`
+provider opentelekomcloud {
+  alias = "region"
+
+  region = "%s"
+}
+`, env.OS_REGION_NAME)
 }
 
 func TestAccPreCheckRequiredEnvVars(t *testing.T) {

--- a/opentelekomcloud/acceptance/s3/resource_opentelekomcloud_s3_bucket_test.go
+++ b/opentelekomcloud/acceptance/s3/resource_opentelekomcloud_s3_bucket_test.go
@@ -948,17 +948,14 @@ resource "opentelekomcloud_s3_bucket" "bucket6" {
 
 func testAccS3BucketConfigWithRegion(randInt int) string {
 	return fmt.Sprintf(`
-provider "opentelekomcloud" {
-  alias  = "reg1"
-  region = "%s"
-}
+%s
 
 resource "opentelekomcloud_s3_bucket" "bucket" {
-  provider = "reg1"
+  provider = "%s"
   bucket   = "tf-test-bucket-%d"
   region   = "%s"
 }
-`, env.OS_REGION_NAME, randInt, env.OS_REGION_NAME)
+`, common.AlternativeProviderWithRegionConfig, common.AlternativeProviderWithRegionAlias, randInt, env.OS_REGION_NAME)
 }
 
 func testAccS3BucketWebsiteConfig(randInt int) string {

--- a/opentelekomcloud/services/s3/resource_opentelekomcloud_s3_bucket.go
+++ b/opentelekomcloud/services/s3/resource_opentelekomcloud_s3_bucket.go
@@ -958,6 +958,8 @@ func resourceS3BucketCorsUpdate(ctx context.Context, s3conn *s3.S3, d *schema.Re
 func resourceS3BucketWebsiteUpdate(ctx context.Context, s3conn *s3.S3, d *schema.ResourceData) error {
 	ws := d.Get("website").([]interface{})
 	switch len(ws) {
+	case 0:
+		return resourceS3BucketWebsiteDelete(ctx, s3conn, d)
 	case 1:
 		var w map[string]interface{}
 		if ws[0] != nil {
@@ -966,8 +968,6 @@ func resourceS3BucketWebsiteUpdate(ctx context.Context, s3conn *s3.S3, d *schema
 			w = make(map[string]interface{})
 		}
 		return resourceS3BucketWebsitePut(ctx, s3conn, d, w)
-	case 2:
-		return resourceS3BucketWebsiteDelete(ctx, s3conn, d)
 	default:
 		return fmt.Errorf("cannot specify more than one website")
 	}

--- a/releasenotes/notes/s3-website-a064467b8b17b337.yaml
+++ b/releasenotes/notes/s3-website-a064467b8b17b337.yaml
@@ -1,0 +1,4 @@
+---
+enhancements:
+  - |
+    **[S3]** Fix error on removing website from ``resource/opentelekomcloud_s3_bucket``

